src/proxy_handler.cc

```diff
@@ -128,7 +202,36 @@ ProxyHandler::HandleRequest(const Request& request, Response* response)
       header.pop_back();
 	  std::string head = header.substr(0, header.find(":"));
 	  std::string value = header.substr(header.find(":") + 2);
-	  response->AddHeader(head, value);
+
+	  if(head == "Cache-Control" 
+	     && value.find("no-store") == std::string::npos 
+	     && value.find("no-cache") == std::string::npos) {
+
+	    std::size_t location_of_max_age = value.find("max-age");
+	    if(location_of_max_age != std::string::npos) {
+	      std::string cache_life = value.substr(location_of_max_age);
+	      std::size_t end_location = cache_life.find(",");
+	      if(end_location != std::string::npos) {
+		cache_life = cache_life.substr(0, end_location);
+	      }
+	    
+	      cache_life = cache_life.substr(8);
+	      boost::posix_time::ptime now = boost::posix_time::second_clock::local_time();
+	      boost::posix_time::time_duration cache_duration(0, 0, std::stoi(cache_life), 0);
+	      boost::posix_time::ptime expiration = now + cache_duration;
+	      if(std::stoi(cache_life) > 0) {
+	      cache_[request.uri()] = expiration;
+	      //Prevents the browser from doing its own cache
```

Comment:
Oh, that's a good point. Should I change it to allow browsers to do their own caches then?